### PR TITLE
add passphrase to init options

### PR DESCRIPTION
**Pull request by richardschneider**
See #194

### CODE REVIEW

File: src/daemon-node.js [hunk 1]
```diff
@@ -117,13 +118,16 @@ class Node {
...
 
-    const keySize = initOpts.keysize || 2048
-
     if (initOpts.directory && initOpts.directory !== this.path) {
       this.path = initOpts.directory
     }
 
-    run(this, ['init', '-b', keySize], { env: this.env }, (err, result) => {
+    const args = ['init', '-b', initOpts.keysize || 2048]
+    if (initOpts.pass) {
```
Comment (dryajov) [inline src/daemon-node.js]: @richardschneider this should already happen since args just get forwarded to the daemon.

**Comment by dryajov** [comment]
@richardschneider I see we already addressed passing the `--pass` options to in-proc nodes here - https://github.com/ipfs/js-ipfsd-ctl/pull/187.

**Comment by richardschneider** [comment]
@dryajov Yes `--pass` works when spawing the daemon.  This PR is to allow it when initing the repo.

**Comment by dryajov** [comment]
@richardschneider `init()` taks an `args` array, that should allow you to pass the `--pass` option already, so no need for that.

**Comment by richardschneider** [comment]
Seems like my **bad**.  So something like the below will add `-pass` to all commands generated to init.

```
const options = {
  args: ['--pass ipfs-is-awesome-software']
}
df.spawn(options, (err, _ipfsd) => { ... })
```

**Comment by dryajov** [comment]
@richardschneider - actually my bad. You're right, init takes an `initOpts` object, and if `pass` is a flag for `init` then we do need this change.

**Comment by richardschneider** [comment]
@dryajov Don't you have PR to fix the jenkins port sharing issue?  If so can we merge it.

**Comment by dryajov** [comment]
@richardschneider yes, it's here  - https://github.com/ipfs/js-ipfsd-ctl/pull/196. 

Just need to make sure that we don't start non `disposable` nodes with the default config (https://github.com/ipfs/js-ipfsd-ctl/pull/196/files#diff-b7c01f8fb93cfc68f6e35c7af488a039R192).  

i.e. 

use something like this when starting non `disposable` nodes:

```
const config = {
            Addresses: {
              Swarm: [
                '/ip4/127.0.0.1/tcp/0/ws',
                '/ip4/127.0.0.1/tcp/0'
              ],
              API: '/ip4/127.0.0.1/tcp/0',
              Gateway: '/ip4/127.0.0.1/tcp/0'
            }
          }
```

**Pull request merged**